com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -312,6 +310,40 @@ define([
                         }
                     }
                 }
+
+                if(form_id == 'end_date' && !$scope.edit_data[role_id].end_date){
+                    $scope.edit_data[role_id].end_date = null;
+                }
+
+                if ($scope.edit_data[role_id].job_contract_id
+                    && $scope.edit_data[role_id].start_date
+                    && typeof $scope.edit_data[role_id].end_date != 'undefined'
+                    && (form_id == 'start_date' || form_id == 'job_contract_id' || form_id == 'end_date')) {
+
+                    $scope.updateRolesWithContractData(role_id);
+                }
+            };
+
+            $scope.updateRolesWithContractData = function (role_id) {
+                var contract_id = $scope.edit_data[role_id].job_contract_id;
+
+                if ($scope.checkIfDatesAreCustom($scope.edit_data[role_id]['start_date'], $scope.edit_data[role_id]['end_date'])) {
+                    var contract = me.contractsData[contract_id];
+
+                    // search for revision containing these dates
+                    var revision = contract.revisions.filter(function (rev) {
+                        return rev.period_start_date == $filter('formatDate')($scope.edit_data[role_id]['start_date'])
+                            && rev.period_end_date == $filter('formatDate')($scope.edit_data[role_id]['end_date']);
+                    });
+
+                    // check if dates match with revision
+                    if (revision.length) {
```

Comment:
Not sure I understand the logic here: any revision is fine in order to override the dates? It doesn't have to be the latest?